invoices/invoiceregistry.go

```diff
@@ -745,46 +756,35 @@ func (i *InvoiceRegistry) NotifyExitHopHtlc(rHash lntypes.Hash,
 
 	// If it isn't recorded, cancel htlc.
 	if !ok {
-		return &HodlEvent{
-			CircuitKey:   circuitKey,
-			AcceptHeight: currentHeight,
-		}, nil
+		return hodlEvent, nil
 	}
 
 	// Determine accepted height of this htlc. If the htlc reached the
 	// invoice database (possibly in a previous call to the invoice
 	// registry), we'll take the original accepted height as it was recorded
 	// in the database.
-	acceptHeight := int32(invoiceHtlc.AcceptHeight)
+	hodlEvent.AcceptHeight = int32(invoiceHtlc.AcceptHeight)
 
 	switch invoiceHtlc.State {
 	case channeldb.HtlcStateCanceled:
-		return &HodlEvent{
-			CircuitKey:   circuitKey,
-			AcceptHeight: acceptHeight,
-		}, nil
+		return hodlEvent, nil
 
 	case channeldb.HtlcStateSettled:
 		// Also settle any previously accepted htlcs. The invoice state
 		// is leading. If an htlc is marked as settled, we should follow
 		// now and settle the htlc with our peer.
-		for key, htlc := range invoice.Htlcs {
+		for _, htlc := range invoice.Htlcs {
```

Comment:
This change looks like it breaks something. There can be multiple htlcs on the invoice with distinct keys. With this change, all keys for the notifications will be the same.